### PR TITLE
fix pretty printer specs

### PR DESCRIPTION
**Pull request by alexbalonperin**
related to https://github.com/DanielaSfregola/random-data-generator/issues/52

### CODE REVIEW

File: src/test/scala/com/danielasfregola/randomdatagenerator/utils/PrettyPrinterSpec.scala [hunk 1]
```diff
@@ -3,12 +3,16 @@ package com.danielasfregola.randomdatagenerator.utils
...
 class PrettyPrinterSpec extends SpecificationLike with Mockito {
 
   abstract class PrettyPrinterSpecContext extends Scope {
+    val logs: ListBuffer[String] = new ListBuffer
+    private def append(log: String): Unit = synchronized { logs += log; () }
+
     val printer = {
-      val mockPrintF = {_: String => (): Unit}
+      val mockPrintF = (log: String) => append(log)
       spy(new PrettyPrinter(mockPrintF))
```
Comment (DanielaSfregola) [inline src/test/scala/com/danielasfregola/randomdatagenerator/utils/PrettyPrinterSpec.scala]: we probably do not a spy anymore. `new PrettyPrinter(mockPrintF)` is enough

**Review by DanielaSfregola — APPROVED** [review]
Hi @zolrag13,
thanks for your contribution!

LGTM! 

Bonus points if you can:
- remove the spy that I believe we do not need
- replicate this exact PR -- the code base is the same! -- for [random-data-generator-magnolia](https://github.com/DanielaSfregola/random-data-generator-magnolia) ;)

**Comment by abalonperin** [comment]
Hey @DanielaSfregola, happy to be able to help!
Didn't have time to get the bonus points, you are too fast! 😄

**Comment by alexbalonperin** [comment]
Hey @DanielaSfregola, happy to be able to help.
I didn't have time to get the bonus points, you are too fast! 😄

**Pull request merged**